File: 007/main.c
#include <stdio.h>
#include <stdlib.h>

int main(void) {
  char *sieve;
  sieve = calloc(10e6, sizeof *sieve);
  size_t prime_count = 0;

  size_t i;

  for (i = 2; i < 10e6; i++) {
    if (!sieve[i]) {
      prime_count++;

      if (prime_count == 10001) {
        printf("%lu\n", i);
        break;
      }

      size_t j;

      for (j = i * 2; j < 10e6; j += i)
        sieve[j] = 1;
    }
  }

  free(sieve);

  return 0;
}

File: 007/main.py
n = int(10e6)
target = 10001
prime_count = 0

sieve = [True] * n
sieve[0] = sieve[1] = False

for i in range(n):
    if sieve[i]:
        prime_count += 1
        if prime_count == target:
            print(i)
            break
        for j in range(i * 2, len(sieve), i):
            sieve[j] = False
